src/test/kotlin/com/buildit/bookit/v1/booking/BookingControllerMockMvcTests.kt

```diff
@@ -33,100 +36,117 @@ class BookingControllerMockMvcTests @Autowired constructor(
     @MockBean
     lateinit var mockRepository: BookingRepository
 
-    @Before
-    fun setupMock() {
-        mockRepository = mock {}
-    }
-    /**
-     * Get a booking
-     */
-    @Test
-    fun getExistingBookingTest() {
-        // arrange
-        // act
-        val result = mockMvc.perform(MockMvcRequestBuilders.get("/v1/booking/1"))
-
-        // assert
-        result.andExpect(MockMvcResultMatchers.status().isOk)
-        result.andExpect(MockMvcResultMatchers.jsonPath<String>("$.subject", Matchers.equalToIgnoringCase("The Booking")))
-    }
-
-    /**
-     * Fail to get a booking
-     */
-    @Test
-    fun getNonexistentBookingTest() {
-        // arrange
-        // act
-        val result = mockMvc.perform(MockMvcRequestBuilders.get("/v1/booking/999"))
-
-        // assert
-        result.andExpect(MockMvcResultMatchers.status().isNotFound)
-    }
-
-    /**
-     * Fail to get a booking
-     */
-    @Test
-    fun getMalformedURITest() {
-        // arrange
-        // act
-        val result = mockMvc.perform(MockMvcRequestBuilders.get("/v1/booking/notanumber"))
-
-        // assert
-        result.andExpect(MockMvcResultMatchers.status().is4xxClientError)
+    @AfterEach
+    fun resetMocks() {
+        reset(mockRepository)
     }
 
-    /**
-     * Create a booking
-     */
-    @Test
-    fun createBookingTest() {
-        // arrange
-        val startDateTime = LocalDateTime.now().plusHours(1)
-        val endDateTime = startDateTime.plusHours(1)
-        val request = BookingRequest(1, "New Meeting", startDateTime, endDateTime)
-
-        Mockito.`when`(mockRepository.insertBooking(1, "New Meeting", startDateTime, endDateTime)).doReturn(Booking(1, 1, "New Meeting", startDateTime, endDateTime))
-        // act
-        val result = mockMvc.perform(MockMvcRequestBuilders.post("/v1/booking")
-            .contentType(MediaType.APPLICATION_JSON)
-            .content(mapper.writeValueAsString(request)))
-
-        // assert
-        result.andExpect(MockMvcResultMatchers.status().`is`(HttpStatus.CREATED.value()))
-        result.andExpect(MockMvcResultMatchers.jsonPath<String>("$.subject", Matchers.equalToIgnoringCase("New Meeting")))
+    @Nested
```

Comment:
I'm curious about the advantage of using @Nested

Reply:
Nested lets you...well... nest tests like you can in Mocha/Expekt.  Lets you keep common before/after functionality w/ the same group of tests.  In this case, I didn't want to repeat the "when" part of the shared mock.  Also, the resetMocks is needed so that tests stay isolated